docs/docs/llms/flow-policy.mdx

```diff
@@ -3,9 +3,179 @@ id: flow-policy
+  alt=""
+/>
+
+The policy allows you to express your logic in flows, shaping the way a conversation 
+progresses based on user input and system response. Other classifiers will
+modify the underlying state machine to handle edge-cases. 
+
+Please refer to the [flows specification](../flows.mdx) for detailed 
+information on defining flows.
+
+## Adding the FlowPolicy to your bot
+
+Incorporating the FlowPolicy into your bot involves a two-step process. First,
+you need to configure the FlowPolicy in your 'config.yml' file. The
+configuration ensures that your bot recognizes and applies the `FlowPolicy` when
+running:
+    
+```yaml-rasa title="config.yml"
+policies:
+# - ...
+  - name: rasa.core.policies.flow_policy.FlowPolicy
+# - ...
+```
+
+Second, you need to add flows to your bot in accordance with the [flows
+specification](../flows.mdx). This step involves defining the states and transitions your bot
+should recognize and respond to.
+
+Once added, the `FlowPolicy` will read all defined flows from your training data
+and apply them.
+
+## How does it work?
+
+The FlowPolicy works by managing the state of your bot, transitioning between
+states, and initiating new flows when required.
+
+### Managing the State
+
+The state of your bot is managed using a 'flow stack'. The flow stack stores the
+states of the conversation, enabling the bot to keep track of where in the flow
+it currently is. The flow stack is stored in the `flow_stack` slot.
+
+Let's take the following example flow and a conversation that is currently
+using it:
+```yaml title="flows.yml"
+flows:
+  transfer_money:
+    description: This flow let's users send money to friends and family.
+
+    steps:
+      - id: "start"
+        intent: transfer_money
+        next: "ask_recipient"
+      - id: "ask_recipient"
+        question: transfer_money_recipient
+        next: "ask_amount"
+      - id: "ask_amount"
+        question: transfer_money_amount_of_money
+        next: "check_transfer_funds"
+    # - ...
+```
+
+<Image
+  img={flowStackEvolution}
+  caption="Changes to the flow stack during a correction of a prior user input"
+  alt=""
+/>
+
+The correction *"oh sorry I meant John"* from the user triggers a correction
+flow to be started. This **correction is not explicitly modeled** in the 
+`transfer_money` flow but rather handled by a so called 
+"conversational pattern".
+
+The correction flow is one of these "conversational patterns". It is only 
+executed when the user corrects a prior input. When triggered, the 
+correction flow is put on top of the flow stack and executed. 
+
+The correction flow is an "interruption" to the `transfer_money` flow. Once the
+correction flow is completed, the `transfer_money` flow is resumed. Before, the
+`transfer_money` is resumed, the `resume_flow` flow is triggered. It allows the
+dialogue designer to define how to guide the user back to the original flow, 
```

Comment:
I'm not sure how it technically allows the dialogue designer to define how to guide the user back to the orginal flow. Can you detail?